arm-redis/2016-04-01/swagger/redis.json

```diff
@@ -0,0 +1,689 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "RedisManagementClient",
+    "description": ".Net client wrapper for the REST API for Azure Redis Cache Management Service",
+    "version": "2016-04-01"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json",
+    "text/json"
+  ],
+  "produces": [
+    "application/json",
+    "text/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Cache/Redis/{name}": {
+      "put": {
+        "tags": [
+          "Redis"
+        ],
+        "operationId": "Redis_CreateOrUpdate",
+        "description": "Create a redis cache, or replace (overwrite/recreate, with potential downtime) an existing cache",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "name",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the redis cache."
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/RedisCreateOrUpdateParameters"
+            },
+            "description": "Parameters supplied to the CreateOrUpdate redis operation."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "201": {
```

Comment:
Are they not findable by dereferencing the $refs?
Tim

Sent from my Windows Phone
________________________________
From: Amar Zavery<mailto:notifications@github.com>
Sent: ‎4/‎29/‎2016 4:00 PM
To: Azure/azure-rest-api-specs<mailto:azure-rest-api-specs@noreply.github.com>
Cc: Tim Lovell-Smith<mailto:tilovell@microsoft.com>; Mention<mailto:mention@noreply.github.com>
Subject: Re: [Azure/azure-rest-api-specs] Adding swagger for redis cache 2016-04-01 api version (#276)


In arm-redis/2016-04-01/swagger/redis.json<https://github.com/Azure/azure-rest-api-specs/pull/276#discussion_r61655626>:

> +            "in": "body",
> +            "required": true,
> +            "schema": {
> +              "$ref": "#/definitions/RedisCreateOrUpdateParameters"
> +            },
> +            "description": "Parameters supplied to the CreateOrUpdate redis operation."
> +          },
> +          {
> +            "$ref": "#/parameters/ApiVersionParameter"
> +          },
> +          {
> +            "$ref": "#/parameters/SubscriptionIdParameter"
> +          }
> +        ],
> +        "responses": {
> +          "201": {


Populate description every where. It forms the basis of documentation.

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub<https://github.com/Azure/azure-rest-api-specs/pull/276/files/edc6f81da2f8c98617d890131d0b1f5f4d61e336#r61655626>